device/iot-device-client/src/main/java/com/microsoft/azure/sdk/iot/device/transport/amqps/AmqpsIotHubConnection.java

```diff
@@ -164,6 +166,21 @@ public void addDeviceOperationSession(DeviceClientConfig deviceClientConfig) thr
         }
     }
 
+    public void reset(ScheduledExecutorService scheduledExecutorService) throws TransportException
```

Comment:
>reset [](start = 16, length = 5)

Could this logic be moved to the open call instead? That way, we only have to close, then open the xxxiothubconection objects

Reply:
if we can, we would also need to move some of the constructor logic into the open call, too

---
In reply to: [258633143](https://github.com/Azure/azure-iot-sdk-java/pull/445#discussion_r258633143) [](ancestors = 258633143)